src/network/networkd-address-label.c

```diff
@@ -404,40 +388,67 @@ int config_parse_address_label(
                 void *data,
                 void *userdata) {
 
-        _cleanup_(address_label_free_or_set_invalidp) AddressLabel *n = NULL;
-        Manager *manager = ltype ? userdata : NULL;
-        Network *network = ltype ? NULL : userdata;
-        uint32_t k;
+        uint32_t k, *label = ASSERT_PTR(data);
         int r;
 
-        assert(filename);
-        assert(section);
-        assert(lvalue);
-        assert(rvalue);
-        assert(userdata);
-
-        r = address_label_new_static(manager, network, filename, section_line, &n);
-        if (r < 0)
-                return log_oom();
-
         if (isempty(rvalue)) {
-                n->label = UINT32_MAX;
-                TAKE_PTR(n);
-                return 0;
+                *label = UINT32_MAX;
+                return 1;
         }
 
         r = safe_atou32(rvalue, &k);
-        if (r < 0) {
-                log_syntax(unit, LOG_WARNING, filename, line, r, "Failed to parse address label, ignoring: %s", rvalue);
-                return 0;
-        }
+        if (r < 0)
+                return log_syntax_parse_error(unit, filename, line, r, lvalue, rvalue);
 
         if (k == UINT32_MAX) {
                 log_syntax(unit, LOG_WARNING, filename, line, 0, "Address label is invalid, ignoring: %s", rvalue);
                 return 0;
         }
 
-        n->label = k;
-        TAKE_PTR(n);
+        *label = k;
+        return 1;
+}
+
+static const ConfigSectionParser address_label_conf_parser_table[_IPV6_ADDRESS_LABEL_CONF_PARSER_MAX] = {
+        [IPV6_ADDRESS_LABEL]        = { .parser = config_parse_ipv6_address_label,        .ltype = 0, .offset = offsetof(AddressLabel, label), },
+        [IPV6_ADDRESS_LABEL_PREFIX] = { .parser = config_parse_ipv6_address_label_prefix, .ltype = 0, .offset = 0,                             },
+};
```

Comment:
maybe make this a local static var inside of config_parse_ipv6_address_label_section()? you don#t need it outside?

could give it a short name than, and communicate that this is a more local concept